webapp/packages/plugin-session-expiration/src/SessionExpireWarningDialog/SessionExpireWarningDialog.tsx

```diff
@@ -20,18 +20,27 @@ import {
 import type { DialogComponent } from '@cloudbeaver/core-dialogs';
 import { ServerNodeChangedDialogStyles } from '@cloudbeaver/plugin-root';
 
-export const SessionExpireWarningDialog: DialogComponent<null, null> = observer(function SessionExpireWarningDialog({ rejectDialog }) {
+interface Payload {
+  onButtonClick: VoidFunction;
+}
+
+export const SessionExpireWarningDialog: DialogComponent<Payload, null> = observer(function SessionExpireWarningDialog({ rejectDialog, payload }) {
   const translate = useTranslate();
   const styles = useS(ServerNodeChangedDialogStyles);
 
+  function onContinueClick() {
+    payload.onButtonClick();
+    rejectDialog();
+  }
+
   return (
     <CommonDialogWrapper size="small" fixedSize>
       <CommonDialogHeader title="app_root_session_expire_warning_title" onReject={rejectDialog} />
       <CommonDialogBody noOverflow>
         <p className={s(styles, { text: true })}>{translate('app_root_session_expire_warning_message')}</p>
       </CommonDialogBody>
       <CommonDialogFooter className={s(styles, { footer: true })}>
-        <Button type="button" mod={['unelevated']} onClick={rejectDialog}>
+        <Button type="button" mod={['unelevated']} onClick={onContinueClick}>
```

Comment:
please revert all related changes
actually when user resolves or closes dialog we will execute `touchSession`
when session expired dialog will be closed automatically and we will check this state